app/controllers/staff/reports_controller.rb

```diff
@@ -125,49 +117,37 @@ def approved_reports(including: [])
   end
 
   def send_csv
-    report_activities = Activity::ProjectsForReportFinder.new(report: @report).call
+    export = Report::Export.new(reports: [@report])
 
-    filename = @report_presenter.filename_for_report_download
-    headers = ExportActivityToCsv.new(report: @report).headers
-
-    stream_csv_download(filename: filename, headers: headers) do |csv|
-      report_activities.each do |activity|
-        csv << ExportActivityToCsv.new(activity: activity, report: @report).call
+    stream_csv_download(filename: export.filename, headers: export.headers) do |csv|
+      export.rows.each do |row|
+        csv << row
       end
     end
   end
 
   def downloadable_reports_for_beis_users
-    report_sets = [
-      active_reports(including: [:organisation]),
-      submitted_reports(including: [:organisation]),
-      in_review_reports(including: [:organisation]),
-      awaiting_changes_reports(including: [:organisation]),
-    ]
-    report_sets.sum.sort_by { |report| report.organisation.name }
-  end
-
-  def report_activities_sorted_by_level(report)
-    Activity::ProjectsForReportFinder.new(
-      report: report,
-      scope: Activity.includes(:organisation, :extending_organisation, :implementing_organisations)
-    ).call.sort_by { |a| a.level }
+    @downloadable_reports_for_beis_users ||= begin
```

Comment:
There seems to be a lot of oddness going on with this export. I'm tempted to leave it, because even a fixed/cleaned up version is going to take forever to export, and @mec's comment in https://trello.com/c/GfKh3G8I/1939-the-download-all-reports-csv-feature-takes-too-long-and-dies suggests we don't even need it anyway